modules/account-lib/test/unit/coin/algo/utils.ts

```diff
@@ -21,4 +22,25 @@ describe('utils', () => {
     const objEncodedByAlgo = algosdk.encodeObj(note);
     should.deepEqual(objEncoded, objEncodedByAlgo);
   });
+
+  it('should build fee data on Pay Tx and Non Participation Tx', () => {
+    const feeRate = 1;
+    const feeDataKeyRegTxn = utils.getFeeData(payTxn, feeRate);
+    // as transactions has less fee than minimum 1000 microalgos feeRate and feeInfo.feeRate are the same
+    should.equal(feeDataKeyRegTxn.feeInfo.feeRate, 4);
+    should.equal(feeDataKeyRegTxn.feeRate, 4);
+    should.equal(feeDataKeyRegTxn.feeInfo.fee, 1000);
+    should.equal(feeDataKeyRegTxn.feeInfo.size, 240);
+
+    const feeDataNonParticipationTxn = utils.getFeeData(nonParticipationTxn, feeRate);
+    should.equal(feeDataNonParticipationTxn.feeInfo.feeRate, 5);
+    should.equal(feeDataNonParticipationTxn.feeRate, 5);
+    should.equal(feeDataNonParticipationTxn.feeInfo.fee, 1000);
+    should.equal(feeDataNonParticipationTxn.feeInfo.size, 207);
+  });
+
+  it('should fail building fee info because fee is negative', () => {
```

Comment:
Add extra tests when fee or feerate are missed values, should throw an excepction.

Reply:
About that. I`ll make the test about feeRate missed and I'll commit it. But I cant handle missed fee here. Because the fee is not on the input of getFeeData. It's made inside using the size calculated.

Reply:
Sounds good let's have a meeting to check that. Thank you. Good Job. :+1:

Reply:
About the feeRate missed I've just checked out that typescript handle this. Because getFeeData() require a number on it's sign. So if you send null or undefined typescript catches and throws the error.